beta-src/src/components/controllers/WDMainController.tsx

```diff
@@ -79,15 +80,16 @@ const WDMainController: React.FC = function ({ children }): React.ReactElement {
   if (!consistentPhase) {
     return <Box>Loading...</Box>;
   }
-
   const showOverlay =
-    noPhase || (displayedPhaseKey && overviewKey !== displayedPhaseKey);
+    noPhase ||
+    status.status === "Left" ||
+    (displayedPhaseKey && overviewKey !== displayedPhaseKey);
   if (displayedPhaseKey === null && overview.phase) {
     setDisplayedPhaseKey(overviewKey);
```

Comment:
I might be missing something with this `Controller` pattern, but I believe this call (and the other dispatch calls) have side effects, which means they would properly be placed inside a `useEffect` or an event handler. I don't think it's right to call them directly in the render function.

Personally, I would put this whole check inside a selector - if you can compute it from the main state, you don't need to store it in separate state. Storing things you can compute is risky because you might forget to update them, or you might get out of sync with the main state in other ways.

Reply:
@lightvector wrote this code, but I believe the reason that this is stateful is that on a phase change, we want to remember the last phase that was being viewed, and let the user manually scroll forward to the current phase. Therefore, it's not a stateless function of the current state.